sycl/source/device_selector.cpp

```diff
@@ -157,5 +160,138 @@ int host_selector::operator()(const device &dev) const {
   return Score;
 }
 
+namespace detail {
+std::string trim_spaces(std::string input) {
+  size_t LStart = input.find_first_not_of(" ");
+  std::string LTrimmed =
+      (LStart == std::string::npos) ? "" : input.substr(LStart);
+
+  size_t REnd = LTrimmed.find_last_not_of(" ");
+  return (REnd == std::string::npos) ? "" : LTrimmed.substr(0, REnd + 1);
+}
+
+std::vector<std::string> tokenize(std::string filter, std::string delim) {
+  std::vector<std::string> Tokens;
+  size_t Pos = 0;
+  std::string Input = filter;
+  std::string Tok;
+
+  while ((Pos = Input.find(delim)) != std::string::npos) {
+    Tok = Input.substr(0, Pos);
+    Input.erase(0, Pos + delim.length());
+    // Erase leading and trailing WS
+    Tok = trim_spaces(Tok);
+
+    if (Tok.size() > 0)
+      Tokens.push_back(Tok);
+  }
+
+  if (Input.size() > 0)
+    Input = trim_spaces(Input);
+
+  // Add remainder
+  if (Input.size() > 0)
+    Tokens.push_back(Input);
+
+  return Tokens;
+}
+
+enum TokenType { TokPlatform, TokDeviceType, TokUnknown };
+
+TokenType parse_kind(std::string token) {
+  TokenType Result = TokUnknown;
+
+  if (token.find("platform") != std::string::npos)
+    Result = TokPlatform;
```

Comment:
You could return TokPlatform right here.
Do you really want to treat "platform_type=cpu" as "type=cpu" ?